torch/csrc/jit/passes/onnx/fold_if_node.cpp

```diff
@@ -0,0 +1,249 @@
+#include <torch/csrc/jit/passes/onnx/fold_if_node.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <torch/csrc/jit/passes/onnx/scalar_type_analysis.h>
+#include <torch/torch.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+// This function determines wheather If Node can be folded.
+static bool isStaticCondition(Node* node) {
+  TORCH_INTERNAL_ASSERT(
+      node->kind() == onnx::If || node->kind() == onnx::Not ||
+      node->kind() == onnx::Identity);
+  auto cast_node = node->input()->node();
+  if (cast_node->kind() != onnx::Cast)
+    cast_node = node;
+  auto prev_node = cast_node->input()->node();
+
+  if (prev_node->kind() == onnx::Not || prev_node->kind() == onnx::Identity ||
+      prev_node->kind() == onnx::If)
+    return isStaticCondition(prev_node);
+
+  auto compare_node = prev_node;
+  if (compare_node->kind() == onnx::Equal ||
+      compare_node->kind() == onnx::Greater ||
+      compare_node->kind() == onnx::Less) {
```

Comment:
onnx function LessOrEqual, GreaterOrEqual?

Reply:
goop point, I have test cases with >= and <= operator, but those get translated as Not Less and Not Greater. I will try to make test cases with LessorEqual and GreaterorEqual

Reply:
actually in opset 12 we have LessOrEqual and GreaterOrEqual